### ISSUE TITLE
BMS AntiAircraft units tagged with Ground+Vehicle

### ISSUE THREAD

**Issue by derammo** [Vyrtuoz/Tacview#9]
I was looking into why BMS 4.37.something AAA and SAM units don't show up correctly in TacView.
BMS writes them out with tags `Type=Ground+Vehicle` because it does not know any more specific info.

Is the following property definition missing something?

```
	<DefaultProperties Id="Core.AntiAircraft" Base="Core.Vehicle">
		<Criteria>
			<Type>AntiAircraft</Type>
			<Name>*</Name>
		</Criteria>
		<Properties>
			<AdditionalType>AntiAircraft</AdditionalType>
			<Shape>AntiAircraft.AAA.ZSU-23-4.obj</Shape>
		</Properties>
	</DefaultProperties>
```

~~Should this have `<AdditionalType>Vehicle+AntiAircraft</AdditionalType>` and would that make it correctly search for SAM and AAA units by name instead of ending up as boxes? :)~~

**Comment by derammo** [comment]
update: ok I just realized I should just try it with XML Default Properties overrides.   But no, that doesn't fix it.

What then would be a fix that would allow something that is tagged as `Ground+Vehicle` to also find units in the `AntiAircraft` subtree?

**Comment by derammo** [comment]
weirdly, I can get the effect that I want by forcing 

```
<?xml version="1.0" encoding="utf-8" standalone="yes" ?>
<DefaultPropertiesCollection LoadingOrder="0.5">
	<DefaultProperties Id="Core.AntiAircraft" Base="Core.Vehicle">
		<Criteria>
			<Type>Ground+Vehicle</Type>
			<Name>*</Name>
		</Criteria>
		<Properties>
			<AdditionalType>AntiAircraft</AdditionalType>
			<Shape>AntiAircraft.AAA.ZSU-23-4.obj</Shape>
		</Properties>
	</DefaultProperties>
</DefaultPropertiesCollection>
```

~~But that doesn't seem right, since none of your examples use the same criteria on multiple tree nodes?    I thought that `<AdditionalType>Ground+Vehicle</AdditionalType>` should do it, but apparently that isn't used for selection criteria?   I don't know enough and will pause here.~~

**Comment by derammo** [comment]
update.   I think I understand now.  With this snippet I am trying to declare that `Core.AntiAircraft` does not require a tag of `AntiAircraft` anymore, only `Vehicle` (inherited) and `Ground`.   So that way it searches that tree for any ground vehicles also.   

```
<?xml version="1.0" encoding="utf-8" standalone="yes" ?>
<DefaultPropertiesCollection LoadingOrder="0.5">
	<DefaultProperties Id="Core.AntiAircraft" Base="Core.Vehicle">
		<Criteria>
			<Type>Ground</Type>
		</Criteria>
	</DefaultProperties>
</DefaultPropertiesCollection>
```

So that pretty much makes stuff work with the current BMS, but it seems like a bad permanent solution, since then `Core.AntiAircraft` units are just generic vehicles.   

If they really don't know (it is campaign-specific) whether things are a generic `Vehicle` or an `AntiAircraft`, then is there a better way to configure it to search?  I don't suppose there is any way to say `Type=(Vehicle|AntiAircraft)+Ground` in the ACMI. :)

By the way, I have been advocating that BMS should offer an option to simply not specify ANY type for the ground units, i.e. no `Type=` at all, because then TacView works beautifully by finding things by name.   This issue and discussion is about if/how to fix it on the TacView side instead, which may just be dumb.

**Comment by BuzyBee1983** [comment]
Hi,

Thanks for all your efforts! Are you a BMS developer or do you otherwise have inside access to the team? We would like to help the team to type the objects correctly. We believe we can help because we have been able to decode VHS files using the correct types. Could you please contact us at support@tacview.net? 

Best regards,

Erin

**Comment by derammo** [comment]
No, I am not a BMS developer, just someone trying to contribute back by running stuff down :).  But I have pointed the correct person at your message via this Discord conversation https://discord.com/channels/263470085560205313/729288056766726144/1296477204430983260
